### ISSUE TITLE
[Azure Stack] az keyvault create throwing ValidationError: The suffix 'mhsm_dns' for this cloud is not set but is used

### ISSUE THREAD

**Issue by rakku-ms** [Azure/azure-cli#15262]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
unable to create keyvault using below cmd for az version >= 2.11.0
az keyvault create throwing ValidationError: The suffix 'mhsm_dns' for this cloud is not set but is used

**To Reproduce**
**Code**
```
$keyvaultName = "testkv"
$resourcegroup = "testkvrg"
$resourcelocation = "redmond"
$environmentName = "ASR"
$armEndpoint = "https://management.redmond.azurestack.corp.microsoft.com/"

$endptres = Invoke-RestMethod "${armEndpoint}/metadata/endpoints?api-version=1.0" -ErrorAction Stop
$portalEndpoint = $endptres.portalEndpoint
$environmentDomainFQDN = $portalEndpoint.Replace($portalEndpoint.Split(".")[0], "").TrimEnd("/").TrimStart(".")
$azureKeyVaultDnsSuffix="vault.$environmentDomainFQDN".ToLowerInvariant()
$storageEndpointSuffix = $environmentDomainFQDN

# Register using CLI
az cloud register --name $environmentName `
        --endpoint-resource-manager $armEndpoint `
        --suffix-storage-endpoint $storageEndpointSuffix `
        --suffix-keyvault-dns ("." + $azureKeyVaultDnsSuffix) `
        --endpoint-vm-image-alias-doc https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-compute/quickstart-templates/aliases.json `
        --profile 2019-03-01-hybrid | Out-Null

az cloud set -n $environmentName | Out-Null
az login | Out-Null
az account set --subscription "5bd2ee47-bec2-411f-8db5-14b67e0920c2" | Out-Null

az group create -n $resourcegroup -l $resourcelocation
$result = az keyvault create --name $keyvaultName --resource-group $resourcegroup --location $resourcelocation --sku standard --enabled-for-deployment true --enabled-for-template-deployment true
```

**Output**
```
az : Switched active cloud to 'ASR'.
At C:\Users\rakku\Desktop\Temp\cli_kv_redmond.ps1:21 char:1
+ az cloud set -n $environmentName | Out-Null
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Switched active cloud to 'ASR'.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
Active subscription switched to 'sarathy (5bd2ee47-bec2-411f-8db5-14b67e0920c2)'.
az : You have logged in. Now let us find all the subscriptions to which you have access...
At C:\Users\rakku\Desktop\Temp\cli_kv_redmond.ps1:22 char:1
+ az login | Out-Null
+ ~~~~~~~~
    + CategoryInfo          : NotSpecified: (You have logged... have access...:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
{
  "id": "/subscriptions/5bd2ee47-bec2-411f-8db5-14b67e0920c2/resourceGroups/testkvrg",
  "location": "redmond",
  "managedBy": null,
  "name": "testkvrg",
  "properties": {
    "provisioningState": "Succeeded"
  },
  "tags": null,
  "type": null
}
az : ValidationError: The suffix 'mhsm_dns' for this cloud is not set but is used.
At C:\Users\rakku\Desktop\Temp\cli_kv_redmond.ps1:26 char:11
+ $result = az keyvault create --name $keyvaultName --resource-group $r ...
+           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (ValidationError...et but is used.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
C:\Users\rakku\.azure\clouds.config may be corrupt or invalid.
Resolve the error or delete this file and try again.
```

**Expected behavior**
 az keyvault create was successfully creating keyvault in Azure Stack environment for az version < 2.11.0

**Environment summary**

```
az --version
azure-cli                         2.12.0

core                              2.12.0
telemetry                          1.0.6

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\rakku\.azure\cliextensions'

Python (Windows) 3.6.8 (tags/v3.6.8:3c6b436a57, Dec 23 2018, 23:31:17) [MSC v.1916 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.

Please let us know how we are doing: https://aka.ms/azureclihats
and let us know if you're interested in trying out our newest features: https://aka.ms/CLIUXstudy
```

**Additional context**

**Comment by rakku-ms** [comment]
@bganapa @sarathys FYI

**Comment by bganapa** [comment]
@Juliehzl , @yonzhan We consume the build from the edge nightly builds in our CI and this is a regression.  We need to fix this before we release the CLI. Could you please investigate?

**Comment by yungezz** [comment]
hi @bganapa are you creating a normal vault or mhsm?

**Comment by yungezz** [comment]
hi @fengzhou-msft @bim-msft one workraround is to add a fake "mhsm_dns" in cloud.json. Does it sound good?

**Comment by Juliehzl** [comment]
Add more information here. 
I think I should expose one additional argument for `az cloud register` to enable the capability for config managed hsm in custom cloud. If you have any concern, feel free to let me know.

@bim-msft, hope we could hide hsm related arguments in the future if it is not supported in Azure Stack Cloud. For the feature we don't support,we shouldn't expose it to customer.

**Comment by fengzhou-msft** [comment]
@Juliehzl before exposing the option, users can add the endpoint in a file for `--cloud-config` with the key `suffix_mhsm_dns` or `suffixMhsmDns`
It's not added in ARM and we need a patch for it.

**Comment by yungezz** [comment]
hi @bim-msft , could the issue be closed since fix released in 2.12.1 already?
